core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -257,12 +257,16 @@ private void updateBalancingState(ConnectivityState state, RoundRobinPicker pick
 
   /**
    * Converts list of {@link EquivalentAddressGroup} to {@link EquivalentAddressGroup} set and
-   * remove all attributes.
+   * remove all attributes except for the channel authority.
    */
   private static Set<EquivalentAddressGroup> stripAttrs(List<EquivalentAddressGroup> groupList) {
     Set<EquivalentAddressGroup> addrs = new HashSet<>(groupList.size());
+
     for (EquivalentAddressGroup group : groupList) {
-      addrs.add(new EquivalentAddressGroup(group.getAddresses()));
+      Attributes whiteListedAttrs = Attributes.newBuilder()
```

Comment:
We do not want to run equals() on Attributes. I added details concerning the approach to take here at https://github.com/grpc/grpc-java/issues/4469#issuecomment-528503518

Reply:
@ejona86, I opted for the second approach you suggested (of keeping a  a Map<stripped EAG, original EAG>), although I did not fully understand the benefit of having the stripping method generic. 
It is private and it isn't used anywhere else. In my opinion the code would become a lot cleaner and more readable if I just changed the stripping method to actually work on EAGs and "do the right thing", i.e. ignore attributes in the comparison. I am happy to refactor it that way if that's desired.

I will also admit that I do not fully understand the implications of calling updateAddresses() on the reused channels as you suggested. You might want to take a particularly close look at how I am doing this.

Reply:
[Attributes.equals() is slow,](https://grpc.github.io/grpc-java/javadoc/io/grpc/Attributes.html#equals-java.lang.Object-) so we are avoiding that.

How are you suggesting to "do the right thing"? Ignore the attributes when comparing within the Map or changing EAG to ignore the attributes in equals/hashCode?